packages/flame_tiled/lib/src/tile_atlas.dart

```diff
@@ -0,0 +1,127 @@
+import 'dart:ui';
+
+import 'package:flame/flame.dart';
+import 'package:flame/image_composition.dart';
+import 'package:flame/sprite.dart';
+import 'package:flame_tiled/src/rectangle_bin_packer.dart';
+import 'package:meta/meta.dart';
+import 'package:tiled/tiled.dart' as tiled;
+
+/// One image atlas for all Tiled image sets in a map.
+class TiledAtlas {
+  /// Single atlas for all renders.
+  // Retain this as SpriteBatch can dispose of the original image for flips.
+  final Image? atlas;
+
+  /// Map of all source images to their new offset.
+  final Map<String, Offset> offsets;
+
+  /// The single batch operation for this atlas.
+  final SpriteBatch? batch;
+
+  /// Image key for this atlas.
+  final String key;
+
+  /// Track one atlas for all images in the Tiled map.
+  TiledAtlas._(this.atlas, this.offsets, this.key)
+      : batch = atlas == null ? null : SpriteBatch(atlas, imageKey: key);
+
+  /// Returns whether or not this atlas contains [source].
+  bool contains(String? source) => offsets.containsKey(source);
+
+  /// Create a new atlas from this object with the intent of getting a new
+  /// [SpriteBatch].
+  TiledAtlas clone() => TiledAtlas._(atlas?.clone(), offsets, key);
+
+  /// Maps of tilesets compiled to [TiledAtlas].
+  @visibleForTesting
+  static final atlasMap = <String, TiledAtlas>{};
+
+  @visibleForTesting
+  static String atlasKey(Iterable<tiled.TiledImage> images) {
+    final files = ([...images.map((e) => e.source)]..sort()).join(',');
+    return 'atlas{$files}';
+  }
+
+  /// Collect images that we'll use in tiles - exclude image layers.
+  static Set<tiled.TiledImage> _onlyTileImages(tiled.TiledMap map) {
+    final imageSet = <tiled.TiledImage>{};
+    for (var i = 0; i < map.tilesets.length; ++i) {
+      final image = map.tilesets[i].image;
+      if (image?.source != null) {
+        imageSet.add(image!);
+      }
+      for (var j = 0; j < map.tilesets[i].tiles.length; ++j) {
+        final image = map.tilesets[i].tiles[j].image;
+        if (image?.source != null) {
+          imageSet.add(image!);
+        }
+      }
+    }
+    return imageSet;
+  }
+
+  /// Loads all the tileset images for the [map] into one [TiledAtlas].
+  static Future<TiledAtlas> loadImages(tiled.TiledMap map) async {
```

Comment:
nit: name this "load" since it's the main public constructor for this class any other name is a bit misleading?

Reply:
I'm going to name it `fromTiledMap` to match `RenderableTiledMap`